### PR TITLE
Major changes to Temporal Anti Aliasing

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
This PR has an almost rewrite of TAA. 

TL;DR: Improved the blurriness and the ghosting of old TAA. 

Main changes (there are more smaller ones, but keeping the list short) 

 - Three tier of quality. Note that higher tier doesn't _necessarily_ mean higher quality for all content, but it means more control to fight bad cases. 

**Neighbourhood clipping**

 - Working space is YCoCg (better neighbourhood clipping behaviour in this space since it decouples luminance).
- Variance clipping. 
- 4 Neighbours or 8 neighbours considered vs previous 2. 

 The above results in better handling of ghosting 
https://drive.google.com/file/d/196y_f5Wm4SYn44bWUpAIZ8a4tAxKzuBr/view?usp=sharing 

**Bicubic resampling** of history with controllable sharpness. This improves with blurriness on movement.  Also added "Anti-ringing" patch option for High preset. 
https://drive.google.com/file/d/1RfexUj1PeyU5blXeHAMsxrvqhhpIj1W5/view?usp=sharing

**Velocity rejection** to deal with situation in which neighbourhood clipping fails. 

**Note**: all the testing has been done on Fontainebleau, spaceship and some artificial scenes. 
I wanted to test more, but the confinement has made that a bit challenging. I expect some feedback coming and some potential additional work will be needed. 

**Documentation will come soon after this lands.**

Yamato: https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/HDRP%252FTAA (failed DX12 passed DX11, not sure is expected?)
DXR tests pass locally.

### CODE REVIEW

**Comment by FrancescoC-unity** [comment]
Momentarily close to do merging as I tried and messed up locally.

**Comment by sebastienlagarde** [comment]
@FrancescoC-unity as you mention, please update doc in another PR. thanks
